src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -1669,7 +1755,7 @@ private static string GetDefaultDataContractNamespace(Type type)
             return ns;
         }
 
-        internal static IList<int> GetDataContractNameForGenericName(string typeName, StringBuilder localName)
+        internal static List<int> GetDataContractNameForGenericName(string typeName, StringBuilder localName)
```

Comment:
Changed this per @justinvp 's comment on https://github.com/dotnet/corefx/pull/13068/files#diff-48232e127d02ac18e342e1847139f290R1838,

> I realize there are a bunch of allocations going on in this method, but one allocation could be avoided by changing the return type of GetDataContractNameForGenericName from IList<int> to List<int> and nestedParamCounts to List<int>, which would avoid the enumerator allocation in the foreach below (since it would use List<T>'s struct enumerator instead of the boxed enumerator that would be returned from IList<int>).